File: src/automatica.core/Automatica.WebNew/src/app/base/model/visu-object-template.ts
import { Model, BaseModel, JsonFieldInfo, JsonProperty, JsonPropertyName } from "./base-model";
import { PropertyTemplate } from "./property-template";

@Model()
export class VisuObjectTemplate extends BaseModel {

    private _MaxWidth: number;
    private _MaxHeight: number;
    private _MinWidth: number;
    private _MinHeight: number;

    @JsonProperty()
    ObjId: string;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    @JsonProperty()
    Key: string;

    @JsonProperty()
    Group: string;

    @JsonProperty()
    Width: number;

    @JsonProperty()
    Height: number;

    @JsonProperty()
    This2VisuPageType: number;

    @JsonPropertyName("PropertyTemplate")
    Properties: PropertyTemplate[] = [];

    @JsonProperty()
    IsVisibleForUser: boolean;

    @JsonProperty()
    public get MaxWidth(): number {
        return this._MaxWidth;
    }
    public set MaxWidth(v: number) {
        this._MaxWidth = v;
    }

    @JsonProperty()
    public get MaxHeight(): number {
        return this._MaxHeight;
    }
    public set MaxHeight(v: number) {
        this._MaxHeight = v;
    }

    @JsonProperty()
    public get MinWidth(): number {
        return this._MinWidth;
    }
    public set MinWidth(v: number) {
        this._MinWidth = v;
    }

    @JsonProperty()
    public get MinHeight(): number {
        return this._MinHeight;
    }
    public set MinHeight(v: number) {
        this._MinHeight = v;
    }

    constructor() {
        super();

    }
    public typeInfo(): string {
        return "VisuObjectTemplate";
    }


    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }

}

File: src/automatica.core/Automatica.Core.EF/Models/VisuObjectTemplate.cs
using System;
using System.Collections.Generic;
using Automatica.Core.Model;

namespace Automatica.Core.EF.Models
{
    public class VisuObjectTemplate : TypedObject
    {
        public VisuObjectTemplate()
        {
            PropertyTemplate = new List<PropertyTemplate>();
        }

        public Guid ObjId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Key { get; set; }
        public string Group { get; set; }
        public float Width { get; set; }
        public float Height { get; set; }

        public float? MaxWidth { get; set; }
        public float? MaxHeight { get; set; }

        public float? MinWidth { get; set; }
        public float? MinHeight { get; set; }

        public long This2VisuPageType { get; set; }

        public bool IsVisibleForUser { get; set; }

        public VisuPageType This2VisuPageTypeNavigation { get; set; }

        public List<PropertyTemplate> PropertyTemplate { get; set; }
    }
}
